src/core/VirtualRenderer.ts

```diff
@@ -314,6 +314,10 @@ export default class VirtualRenderer {
                 }
             }
         }
+
+        // Future calls to get stable IDs must use the function
+        // provided by the new data provider.
+        this._fetchStableId = getStableId;
```

Comment:
`_fetchStableId` always refers the the latest dataProvider. Check here: https://github.com/Flipkart/recyclerlistview/blob/master/src/core/RecyclerListView.tsx#L157 . What issue were you noticing?

Reply:
This is in the function used to sync the virtual renderer with a new data provider. The virtual renderer keeps a reference to a getStableId function which must be updated when the data provider changes. If this is not done, then it will fail to correctly associate indexes with stableIds when the new data provider returns different stableIds for each index - e.g. if an item is inserted at the start of the data array.

I did not find any other place in the code where _fetchStableId could be updated when the data provider is changed, and this seems like the best place to do it.

Reply:
Hi,

You can test this bug by the following steps:
1. Instantiate a RecyclerListView with a DataProvider containing 1 item.
2. Change the DataProvider to one containing 3 items
3. The RecyclerListView will crash due to trying to look up the 2 new items using the old DataProvider's getStableId function.

Thanks.